### PR TITLE
Bump Nessie from 0.18.0 to 0.19.0

### PR DESCRIPTION
**Pull request by snazy**

### CODE REVIEW

**Review by nastra — APPROVED** [review]

File: versions.props [hunk 1]
```diff
@@ -24,7 +24,7 @@ javax.activation:activation = 1.1.1
 org.glassfish.jaxb:jaxb-runtime = 2.3.3
 software.amazon.awssdk:* = 2.15.7
 org.scala-lang:scala-library = 2.12.10
-org.projectnessie:* = 0.18.0
+org.projectnessie:* = 0.19.0
```
[the remark this answers is not in the document]
Comment (ajantha-bhat) [reply versions.props:27]: oh.. I see that site docs are moved. `mkdocs.yml` is not in this repo.
It is even hard to track it now :)

File: nessie/src/test/java/org/apache/iceberg/nessie/BaseTestIceberg.java [hunk 2]
```diff
@@ -102,9 +104,9 @@ private void resetData() throws NessieConflictException, NessieNotFoundException
   }
 
   @BeforeEach
-  public void beforeEach() throws IOException {
-    uri = server.getURI().toString();
-    this.api = HttpClientBuilder.builder().withUri(uri).build(NessieApiV1.class);
+  public void beforeEach(@NessieUri URI nessieUri) throws IOException {
```
Comment (kbendick) [inline nessie/src/test/java/org/apache/iceberg/nessie/BaseTestIceberg.java:107]: Question: What does this do / how does this work? Anything you can point me to (PR, etc) so I can peruse it myself?

**Review by kbendick — APPROVED** [review]
LGTM.

I think there's a nessie version in the site docs (or there was). So you might want to make a note to follow up on that. They just underwent a big refactor (and are still undergoing finishing touches on that) so you might have to wait depending but FYI. 🙂

File: nessie/src/test/java/org/apache/iceberg/nessie/BaseTestIceberg.java (hunk 2 above)
Comment (snazy) [reply nessie/src/test/java/org/apache/iceberg/nessie/BaseTestIceberg.java:107]: That was part of some refactoring for the test-related code, actually: make Jersey correctly expose the "right" base URI, as there was a bug when multiple tests using that Jersey-based code ran concurrently.
Previously there was a getter-method on the "server" class, but that became a parameter being filled from the JUnit5 extension (hence the `@NessieUri` annotation).

Comment (nastra) [reply nessie/src/test/java/org/apache/iceberg/nessie/BaseTestIceberg.java:107]: just for completeness for @kbendick, this was introduced in https://github.com/projectnessie/nessie/pull/3073

**Review by rymurr — APPROVED** [review]

**Comment by ajantha-bhat** [comment]
Handled the site doc update.
https://github.com/apache/iceberg-docs/pull/57

**Pull request merged**